### ISSUE TITLE
The difference with Adam and AdamP in the code

### ISSUE THREAD

**Issue by mbax4ds5** [clovaai/AdamP#16]
Hello,

I am just following the major differences between Adam and AdamP.
I can just notice the difference where it says #Projection in the code.

I have been running some baseline cases with AdamP excluding that projection part and pytorch Adam.
However, the results look somehow different.

Is there any part that I am missing?
(Initial states are fixed the same, seeds are the same too!)

Thanks!

**Comment by bhheo** [comment]
Hi @mbax4ds5 

Thank you for your interest in our research.

First of all, `AdamP` is based on `AdamW`, not `Adam`
I think it is the most significant difference.

If you used Nesterov gradients `nesterov=True`, it is different from `AdamW` because `AdamW` doesn't have the Nesterov option.
To the best of my knowledge, except for Nesterov and projection, all parts of `AdamP` are the same as `AdamW`.

**Comment by mbax4ds5** [comment]
@bhheo 
Hi, thanks for the quick reply!
I have been trying to modify some bit in AdamP.
I understand it is not different from AdamW in the conditions you said earlier.
As far as I understand the difference between Adam and AdamW is the weight decay part.
So I modified AdamP to simply see if AdamP with no Projection and Decay performs the same as Adamp but it did not.
I am just curious if I made a mistake or there is some extra bit which is not part of original Adam.

I know it is slightly off the topic of AdamP but I would appreciate your reply :)
Thanks!
